.github/workflows/build.yml

```diff
@@ -162,17 +109,38 @@ jobs:
           retry_wait_seconds: 10
           command: docker pull docker.pkg.github.com/apache/incubator-nuttx-testing/nuttx-ci-linux
 
-      - name: Export NuttX Repo SHA
-        run:  echo "::set-env name=nuttx_sha::`git -C sources/nuttx rev-parse HEAD`"
-      - name: Refresh Git Credentials
+      - uses: actions/download-artifact@v2
+        with:
+          name: gitrefs
+          path: gitrefs-rt/
+      - name: Export Repo Refs
+        run: source gitrefs-rt/gitenv.sh
+
+      - name: Checkout nuttx repo
         uses: actions/checkout@v2
         with:
           repository: apache/incubator-nuttx
-          ref: ${{ env.nuttx_sha }}
+          ref: ${{ env.os_ref }}
           path: sources/nuttx
           fetch-depth: 1
-      - name: Get Tags for NuttX Repo
-        run: git -C sources/nuttx fetch --tags
+      - run: git -C sources/nuttx pull --tags
```

Comment:
Should we use "git -C sources/nuttx fetch --prune --unshallow" to get both branch and tag? I am afraid that we may want the branch at the late time.

Reply:
We should consider that this might not be a branch and instead the merge ref that is being tested (that's what you get by default with the checkout action if no ref is explicitly set). This matches the existing functionally so I don't see  a reason to change it.

Reply:
Ok.